### PR TITLE
introduce from_rserve method for request object

### PR DESCRIPTION
**Pull request by dselivanov**

### CODE REVIEW

File: R/RestRserveResponse.R [hunk 1]
```diff
@@ -159,28 +159,33 @@ RestRserveResponse = R6::R6Class(
     encode = NULL,
     #------------------------------------------------
     initialize = function(body = "",
-                          content_type = 'text/plain',
-                          headers = NULL,
+                          content_type = "text/plain",
+                          headers = list_named(),
                           status_code = 200L,
                           encode = NULL) {
       if (isTRUE(getOption('RestRserve_RuntimeAsserts', TRUE))) {
```
Comment (artemklevtsov) [inline R/RestRserveResponse.R]: Seems now this block not in runtime.

File: R/RestRserveResponse.R [hunk 2]
```diff
@@ -159,28 +159,33 @@ RestRserveResponse = R6::R6Class(
...
-      }
+      self$headers = headers
       self$status_code = as.integer(status_code)
       self$cookies = list()
       self$context = new.env(parent = emptyenv())
       self$encode = encode
     },
+    reset = function() {
+      self$body = ""
+      self$set_content_type("text/plain")
```
Comment (artemklevtsov) [inline R/RestRserveResponse.R]: Should be taken from the app class field or global option.

File: R/RestRserveResponse.R (hunk 1 above)
Comment (dselivanov) [reply R/RestRserveResponse.R]: not runtime (so the 'RestRserve_RuntimeAsserts' may be misleading), but users will still call `new` method. Hence we need asserts

File: R/RestRserveResponse.R (hunk 2 above)
Comment (dselivanov) [reply R/RestRserveResponse.R]: very good catch! thanks

File: R/RestRserveResponse.R (hunk 1 above)
Comment (artemklevtsov) [reply R/RestRserveResponse.R]: Sure. I mean that we can remove `if` condition.

File: R/utils.R [hunk 3]
```diff
@@ -37,3 +37,8 @@ is_string = function(x) {
 is_path = function(path) {
   is_string(path) && startsWith(path, "/")
 }
+
+list_named = function(length = 0, names = character(0)) {
```
Comment (artemklevtsov) [inline R/utils.R]: Must be tested in tests.

**Pull request merged**